nuxeo-features/preview/nuxeo-preview-core/src/main/java/org/nuxeo/ecm/platform/preview/adapter/base/ConverterBasedHtmlPreviewAdapter.java

```diff
@@ -103,6 +109,10 @@ public void setDefaultPreviewFieldXPath(String xPath) {
     @Override
     public boolean hasPreview(String xpath) {
         String srcMT = getMimeType(xpath);
+        if (!Framework.getService(ConfigurationService.class).isBooleanPropertyTrue(ALLOW_ZIP_PREVIEW)
+                && "application/zip".equals(srcMT)) {
```

Comment:
Put `"application/zip".equals(srcMT)` first before the `&&` as it's cheaper to evaluate.